### PR TITLE
Support disabling mutators through Profile fields

### PR DESCRIPTION
**Pull request by 0xedward**
This PR allows mutators to be disabled when the name of the mutator is included in a profile's `disabledMutators` field. Disabling mutators is helpful for:
1. running instances of Fuzzilli with different mutators without having to maintain/build two separate Fuzzilli binaries
2. experimenting with a subset of mutators
3. tailoring Fuzzilli for a target (e.g. JS engines without JIT compilers can disable `JITStressMutator` without maintaining a fork of Fuzzilli or JS engines that have not implemented language features used in `JavaScriptExploreLifting.swift` can disable `ExplorationMutator`)

### Test plan
1. Add print statement to print out list of mutators in `main.swift`
4. Add any mutator into `disabledMutators` field for any profile (e.g. `JITStressMutator`)
5. Run Fuzzilli and check if mutator is present in the list of mutators printed out
```swift
WeightedList<Mutator>(elements: [(elem: Fuzzilli.ExplorationMutator, weight: 3, cumulativeWeight: 0), (elem: Fuzzilli.CodeGenMutator, weight: 2, cumulativeWeight: 3), (elem: Fuzzilli.SpliceMutator, weight: 2, cumulativeWeight: 5), (elem: Fuzzilli.ProbingMutator, weight: 2, cumulativeWeight: 7), (elem: Fuzzilli.InputMutator, weight: 2, cumulativeWeight: 9), (elem: Fuzzilli.InputMutator, weight: 1, cumulativeWeight: 11), (elem: Fuzzilli.OperationMutator, weight: 1, cumulativeWeight: 12), (elem: Fuzzilli.CombineMutator, weight: 1, cumulativeWeight: 13)], totalWeight: 14)
```

### Example usage
```swift
// HermesProfile.swift
let hermesProfile = Profile(
    ...
    disabledMutators: ["JITStressMutator"],
    ...
)
```

```swift
// HermesExperimentalProfile.swift
let hermesProfile = Profile(
    ...
    // Fuzzilli instance focused on variant finding
    disabledMutators: ["JITStressMutator", "CodeGenMutator", "SpliceMutator", "CombineMutator"],
    ...
)
```

### CODE REVIEW

**Review by carl-smith — CHANGES_REQUESTED** [review]
Thank you Edward! :)

File: Sources/FuzzilliCli/main.swift [hunk 1]
```diff
@@ -432,7 +433,7 @@ func makeFuzzer(with configuration: Configuration) -> Fuzzer {
         // (ConcatMutator(),                1),
         (OperationMutator(),                1),
         (CombineMutator(),                  1),
-    ])
+    ]).filter({ !disabledMutators.contains(String(describing: type(of: $0))) })
```
[the remark this answers is not in the document]
Comment (0xedward) [reply Sources/FuzzilliCli/main.swift]: Oh, that's a good idea! I added a check for that

File: Sources/FuzzilliCli/main.swift [hunk 2]
```diff
@@ -368,6 +370,35 @@ func makeFuzzer(with configuration: Configuration) -> Fuzzer {
...
+        // (ConcatMutator(),                1),
+        (OperationMutator(),                1),
+        (CombineMutator(),                  1),
+        // Include this once it does more than just remove unneeded try-catch
+        // (FixupMutator()),                1),
+    ])
+    let mutatorsSet = Set(mutators.map {String(describing: type(of: $0))})
+    if !disabledMutators.isSubset(of: mutatorsSet) {
+        configError("The following mutators in " + profileName + " profile's disabledMutators do not exist: " + disabledMutators.subtracting(_: mutatorsSet).description + ". Please check and remove them from your profile configuration.")
+    }
```
Comment (0xedward) [inline Sources/FuzzilliCli/main.swift:392]: Currently, if there's any mutators in the specified profile's `disabledMutators` that aren't in `mutators`, then we print this message and exit.

Do you think we should exit after informing the user or let Fuzzilli keep executing (i.e. switching from using `configError` to `logger.info`)?

**Review by 0xedward — COMMENTED** [review]
Thanks for the review, Carl! Sorry again about the delay in updating this PR with your suggestions

**Comment by 0xedward** [comment]
As a note: The CI build job failures seem to be related to https://github.com/googleprojectzero/fuzzilli/commit/292aa0b97fa25b9af413d5a7c06e7bb9ac8556c3 and not to the changes in this PR

File: Sources/FuzzilliCli/main.swift (hunk 2 above)
Comment (carl-smith) [reply Sources/FuzzilliCli/main.swift:392]: I think this should be a `configError`

File: Sources/FuzzilliCli/main.swift [hunk 3]
```diff
@@ -368,6 +370,35 @@ func makeFuzzer(with configuration: Configuration) -> Fuzzer {
...
+        // Can be enabled for experimental use, ConcatMutator is a limited version of CombineMutator
+        // (ConcatMutator(),                1),
+        (OperationMutator(),                1),
+        (CombineMutator(),                  1),
+        // Include this once it does more than just remove unneeded try-catch
+        // (FixupMutator()),                1),
+    ])
+    let mutatorsSet = Set(mutators.map {String(describing: type(of: $0))})
+    if !disabledMutators.isSubset(of: mutatorsSet) {
+        configError("The following mutators in " + profileName + " profile's disabledMutators do not exist: " + disabledMutators.subtracting(_: mutatorsSet).description + ". Please check and remove them from your profile configuration.")
```
Comment (carl-smith) [inline Sources/FuzzilliCli/main.swift:391]: I think you can call it like this:
`configError("The following mutators in \(profileName) profile's disabledMutators do not exist: \(disabledMutators.subtracting(mutatorsSet)).")`

File: Sources/FuzzilliCli/main.swift [hunk 4]
```diff
@@ -368,6 +370,35 @@ func makeFuzzer(with configuration: Configuration) -> Fuzzer {
...
+        (InputMutator(isTypeAware: false),  2),
+        (InputMutator(isTypeAware: true),   1),
+        // Can be enabled for experimental use, ConcatMutator is a limited version of CombineMutator
+        // (ConcatMutator(),                1),
+        (OperationMutator(),                1),
+        (CombineMutator(),                  1),
+        // Include this once it does more than just remove unneeded try-catch
+        // (FixupMutator()),                1),
+    ])
+    let mutatorsSet = Set(mutators.map {String(describing: type(of: $0))})
```
Comment (carl-smith) [inline Sources/FuzzilliCli/main.swift:389]: You should be able to do this here I think:
`Set(mutators.map { $0.name })`

**Comment by carl-smith** [comment]
Thank you so much Edward! :) 
Yes, we are aware of the CI failures, we are ignoring them for now.

**Pull request merged**